### PR TITLE
#16 CLI to help generate Dockerfile based on features.

### PR DESCRIPTION
**Pull request by last-ent**
CLI is implemented to a list of features and then create a Dockerfile.
Building a docker image is a more complicated item which has a separate ticket for it.

Current implementation provides a substitution that will only create a dockerfile.

### CODE REVIEW

File: cli/src/pazuzu_cli/console.clj [hunk 1]
```diff
@@ -13,10 +13,40 @@
...
                     (str (clojure.string/join "" path) "/Dockerfile"))]
     (spit save-path docker-file-data)
     save-path))
+
+
+(def feature-flags {"-f" :features "-p" :path "--dry-run" :dry-run})
+
+
+(defn to-args-map
+  "Convert the list of args into a Hash Map charting all the arguments & flags.
```
Comment (dryewo) [inline cli/src/pazuzu_cli/console.clj]: By the way, there is https://github.com/clojure/tools.cli
No need to reinvent the wheel :)

File: cli/src/pazuzu_cli/core.clj [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
 
 
+(defn create-dockerfile
+  [& [features]]
+  (println "Dockerfile created @ "
+           (->> features console/fetch-and-compile-features console/save-dockerfile)))
+
+
+; TODO: STUB!! Rewrite this function once we have function that will actually build a dockerfile.
+(def build-docker-image create-dockerfile)
```
Comment (dryewo) [inline cli/src/pazuzu_cli/core.clj]: You can use this code: https://github.com/dryewo/docker-maker/blob/master/src/docker_maker/docker.clj

**Comment by dryewo** [comment]
Please consider using existing solutions:
- clojure.cli library
- old docker-maker project

**Comment by dryewo** [comment]
👍

**Comment by mikkeloscar** [comment]
@dryewo I don't assume docker-maker is in maven. Is it possible to import it directly from github, or how would you go about using it in pazuzu?

File: cli/src/pazuzu_cli/console.clj (hunk 1 above)
Comment (last-ent) [inline cli/src/pazuzu_cli/console.clj]: I tried using tools.cli & cli4clj but it wasn't working as expected. I will have a look at it again, maybe I missed some config and raise another PR for it.

File: cli/src/pazuzu_cli/core.clj (hunk 2 above)
Comment (last-ent) [inline cli/src/pazuzu_cli/core.clj]: 👍

**Comment by dryewo** [comment]
@mikkeloscar regarding docker-maker I would suggest just copying the source code :)

**Comment by mikkeloscar** [comment]
Ok, I'll do that then. :)

**Pull request merged**